app/src/main/java/com/github/houseorganizer/houseorganizer/MainScreenActivity.java

```diff
@@ -42,9 +47,45 @@ protected void onCreate(Bundle savedInstanceState) {
         setContentView(R.layout.activity_main_screen);
 
         mAuth = FirebaseAuth.getInstance();
+        mUser = mAuth.getCurrentUser();
         db = FirebaseFirestore.getInstance();
-        // Test household for now because house buttons don't do anything
-        currentHouse = db.document("households/test_house_0");
+
+        Intent intent = getIntent();
+        String householdId = intent.getStringExtra(HOUSEHOLD);
+        TextView text = findViewById(R.id.last_button_activated);
+        currentHouse = null;
+
+        if (householdId != null) {
+            // Current selected house
+            currentHouse = db.collection("households").document(householdId);
+            text.setText("currentHouse: " + currentHouse.getId());
+
+        } else {
+            // Current house by default
+            // Cannot test still, need to sign in first
+            /*db.collection("households")
+                    .whereArrayContains("residents", mUser.getUid())
+                    .get()
+                    .addOnCompleteListener(task -> {
+                        if (task.isSuccessful()) {
+                            ArrayList<String> households = new ArrayList<String>();
+                            for (QueryDocumentSnapshot document : task.getResult()) {
+                                String id = document.getId();
+                                households.add(id);
+                            }
+
+                            currentHouse = db.collection("households").document(households.get(0));
+                            text.setText("currentHouse: " + currentHouse.getId() + " by default");
+
+                        } else {
+                            Toast.makeText(getApplicationContext(), "Could not get a house.", Toast.LENGTH_SHORT).show();
+                        }
+                    });*/
```

Comment:
Is the user not signed in by the time he gets on this activity ? We can probably make it so that if no houses are found fall back to the temp case for now

Reply:
For this activity the user should be signed in. If no houses are found that means that there was an error trying to retrieve the info from firebase, therefore I think showing an error message is best.